File: e2e/cypress-javascript/context/TaskContext.js
const { TaskPriority } = require('@/data/TaskPriority');
const { TaskStatus } = require('@/data/TaskStatus');

class TaskContext {
  constructor(overrides = {}) {
    this.id = overrides.id ?? crypto.randomUUID();
    this.title = overrides.title ?? `Task-${crypto.randomUUID().split('-')[0]}`;
    this.description = overrides.description ?? 'Automated test task description';
    this.status = overrides.status ?? TaskStatus.TODO;
    this.priority = overrides.priority ?? TaskPriority.MEDIUM;
  }

  createTaskRequest() {
    return {
      title: this.title,
      description: this.description,
      status: this.status,
      priority: this.priority,
    };
  }

  createTaskData() {
    return {
      title: this.title,
      description: this.description,
      status: this.status,
      priority: this.priority,
    };
  }

  createTaskResponse() {
    return {
      id: this.id,
      title: this.title,
      description: this.description,
      status: this.status,
      priority: this.priority,
    };
  }
}

module.exports = { TaskContext };

File: e2e/capybara-ruby/context/task_context.rb
# frozen_string_literal: true

require 'securerandom'

require_relative '../data/task_priority'
require_relative '../data/task_status'

class TaskContext
  attr_reader :id, :title, :description, :status, :priority

  def initialize(overrides = {})
    @id = overrides[:id] || SecureRandom.uuid
    @title = overrides[:title] || "Task-#{SecureRandom.uuid.split('-').first}"
    @description = overrides[:description] || 'Automated test task description'
    @status = overrides[:status] || TaskStatus::TODO
    @priority = overrides[:priority] || TaskPriority::MEDIUM
  end

  def create_task_request
    {
      title: title,
      description: description,
      status: status,
      priority: priority
    }
  end

  def create_task_data
    {
      title: title,
      description: description,
      status: status,
      priority: priority
    }
  end

  def create_task_response
    {
      id: id,
      title: title,
      description: description,
      status: status,
      priority: priority
    }
  end
end
